### ISSUE TITLE
First line of deployment reports are displaying incorrectly

### ISSUE THREAD

**Issue by haltman-at** [trufflesuite/truffle#4679]
## Issue

Something's up wit the first lines of deployment reports; see example below.

## Steps to Reproduce

Try deploying a contract, that should suffice!

## Actual Results

Every deployment report looks like this at the start:

```
1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   ⠋ Blocks: 0            Seconds: 0   > transaction hash:    0x11f1e20b9fe2a3d465a517356ecb2eafe5c6ef70a85747ac43beabf6c31e3f22
   > Blocks: 0            Seconds: 0
   > contract address:    0x513Dfe39C05BFbEcb22AF30e681D1e08C2256A2b
```

The rest of the report is fine.  Note that this error occurs for each contract deployed, not just the first one, nor just the first one in a file.

## Environment

* Truffle version (`truffle version`): 5.4.31
* node version (`node --version`): 12.22.9

**Comment by lsqproduction** [comment]
Is it the formating of the report - how transaction hash is on the side instead of below Blocks and Seconds?
Or is it something else?

**Comment by haltman-at** [comment]
Yes, the transaction hash is on the side; but also, blocks and seconds appears twice; and moreover, the first time it appears, it has an incomplete spinner next to it, rather than a `>`.

**Comment by cds-amal** [comment]
@eggplantzzz also pointed out that there's extra crud at the end of migration.

**Comment by lsqproduction** [comment]
Did this got fixed?
my initial migration looks pretty. Or are there other reproduction steps? I did truffle develop and then migrate.
```
1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0xad5f99688f993f3c04f1020171329bfc158e449c6713fdb8c79f0f6a6ebc4a4c
   > Blocks: 0            Seconds: 0
   > contract address:    0x8c234caaE6Dd6bB7E189FA20Aa9440D358ee5164
   > block number:        1
   > block timestamp:     1643997505
   > account:             0x512292D7C303E288dAeFB68607f8ae9b7A972DBE
   > balance:             99.999441521875
   > gas used:            165475 (0x28663)
   > gas price:           3.375 gwei
   > value sent:          0 ETH
   > total cost:          0.000558478125 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:      0.000558478125 ETH
```

the end of migration is still ugly. 

```
Summary
=======
> Total deployments:   3
> Final cost:          0.00175477248451957 ETH


- Blocks: 0            Seconds: 0
- Saving migration to chain.
- Blocks: 0            Seconds: 0
- Blocks: 0            Seconds: 0
- Saving migration to chain.
- ```

**Comment by haltman-at** [comment]
> my initial migration looks pretty. Or are there other reproduction steps? I did truffle develop and then migrate.

Hm, yes, when I do it inside `truffle console` or `truffle develop`, the beginning looks fine.  However the problem occurs consistently when I simply run `truffle migrate` at the command line, rather than inside the Truffle Console.

**Comment by lsqproduction** [comment]
> > my initial migration looks pretty. Or are there other reproduction steps? I did truffle develop and then migrate.
> 
> Hm, yes, when I do it inside `truffle console` or `truffle develop`, the beginning looks fine. However, the problem occurs consistently when I simply run `truffle migrate` at the command line, rather than inside the Truffle Console.

got it. thank you. When doing Truffle migrate I can see the error in the beginning, and while using truffle console or truffle develop, the error is at the end of the output.

**Comment by haltman-at** [comment]
Oh hey I just noticed another problem here.  Here's what shows up in dry-run:

```
1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
undefined
   > block number:        6379579
```

There's that `undefined` printed that shouldn't be... meanwhile, blocks, transaction hash, and contract address are all missing!

**Comment by haltman-at** [comment]
Oh dear, this is what it looks like when I migrate something to Goerli for real:

```
1_initial_migration.js
======================

   Replacing 'Migrations'
   ----------------------
   ⠋ Blocks: 0            Seconds: 0   > transaction hash:    0x85ebc0059c6c24e976342e6bdbe3e33bfef9c77192e911f51359953984ea2e54
   ⠹ Blocks: 0            Seconds: 0undefined
   ⠹ Blocks: 1            Seconds: 4undefined
   ⠙ Blocks: 1            Seconds: 8undefined
   ⠙ Blocks: 1            Seconds: 12undefined
   > Blocks: 2            Seconds: 16
   > contract address:    0x47dF5CEDAC7a9AbBc73c6dc21525a60aF7aF7127
   > block number:        6379646
```

**Comment by gnidan** [comment]
This is now fixed and released in [Truffle v5.5.1](https://github.com/trufflesuite/truffle/releases/tag/v5.5.1)
